pkg/arvo/app/hark-store.hoon

```diff
@@ -0,0 +1,197 @@
+::  hark-store: notifications [landscape]
+::
+/-  store=hark-store, post, group-store, metadata-store
+/+  resource, metadata, default-agent, dbug, graph-store
+::
+~%  %hark-store-top  ..is  ~
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-0
+  ==
+::
++$  state-0
+  $:  %0
+      =notifications:store
+      archive=notifications:store
+      last-seen=@da
+  ==
+::
+++  orm  ((ordered-map @da timebox:store) lth)
+--
+::
+=|  state-0
+=*  state  -
+::
+=<
+%-  agent:dbug
+^-  agent:gall
+~%  %hark-store-agent  ..card  ~
+|_  =bowl:gall
++*  this  .
+    ha    ~(. +> bowl)
+    def   ~(. (default-agent this %|) bowl)
+    met   ~(. metadata bowl)
+::
+++  on-init
+  :_  this
+  ~[autoseen-timer]
+::
+++  on-save  !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  `this(state !<(state-0 old))
+::
+++  on-watch  on-watch:def
+++  on-peek   on-peek:def
+++  on-poke
+  ~/  %hark-store-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  ?>  (team:title our.bowl src.bowl)
+  =^  cards  state
+    ?+  mark           (on-poke:def mark vase)
+        %hark-action   (hark-action !<(action:store vase))
+    ==
+  [cards this]
+  ::
+  ++  hark-action
+    |=  =action:store
+    ^-  (quip card _state)
+    |^
+    ?-  -.action
+      %add      (add +.action)
+      %archive  (do-archive +.action)
+      %seen     seen
+      %read     (read +.action)
+      %unread   (unread +.action)
+    ==
+    ++  add  
+      |=  [=index:store =notification:store]
+      ^-  (quip card _state)
+      =/  =timebox:store
+        (gut-orm:ha notifications last-seen)
+      =/  existing-notif
+        (~(get by timebox) index)
+      =/  new=notification:store
+        ?~  existing-notif
+          notification
+        (merge-notification:ha u.existing-notif notification)
+      =.  timebox
+        (~(put by timebox) index new)
+      =.  notifications
+        (put:orm notifications last-seen timebox)
+      :_(state (give:ha [/updates]~ %add index notification))
+    ::
+    ++  do-archive
+      |=  [time=@da =index:store]
+      ^-  (quip card _state)
+      =/  =timebox:store
+        (gut-orm:ha notifications time)
+      =/  =notification:store
+        (~(got by timebox) index)
+      =.  timebox
```

Comment:
Stylistically, I think it might be clearer to perform this state mutation inside the later `%_  state  ...`, so there's only one place to look to see how the state gets updated.  You could `=/  new-timebox` here, and use that value in the `%_`.

The shadowing of `timebox` in the `+jub-orm` call in the `%_` is similarly confusing.